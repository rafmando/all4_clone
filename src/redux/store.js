import { configureStore } from '@reduxjs/toolkit'
import watchLaterReducer from './watchLaterSlice'

export const store = configureStore({
  reducer: {
    watchLaterReducer: watchLaterReducer,
  },
})