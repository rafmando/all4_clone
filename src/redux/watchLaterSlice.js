import { createSlice } from '@reduxjs/toolkit'

const initialState = {
    watchLater:[]
}

const watchLaterSlice = createSlice({
    name: 'watchLater',
    initialState,
    reducers: {
        addToWatchLater: (state,action) => {
            const show = {
                id: action.payload.id,
                name: action.payload.name,
                img: action.payload.img,
                description: action.payload.description,
            }

            const quantity = action.payload.name.split
            console.log(quantity)
            state.watchLater.push(show)
            
            
        }
    }
});

export const {
    addToWatchLater
} = watchLaterSlice.actions
export default watchLaterSlice.reducer