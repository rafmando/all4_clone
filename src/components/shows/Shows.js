import { useState, useEffect }from 'react'
import ShowsStyles from './Shows.styles'
import { IoIosArrowDown } from 'react-icons/io'
import showsData from '../../data/ShowsData'
import { useDispatch } from 'react-redux'
import { addToWatchLater } from '../../redux/watchLaterSlice'


const {
    StyledShowsContainer,
    StyledShowsWrapper,
    StyledShowsHeader,
    StyledShowsTitleWrapper,
    StyledShowsTitle,
    StyledShowCount,
    StyledShowsRefine,
    StyledShowsRefineIcon,
    StyledShowsGenresWrapper,
    StyledShowsGenre,
    StyledShowsMoreBox,
    StyledShows,
    StyledShowsCard,
    StyledShowsCardOverlay,
} = ShowsStyles

const Shows = () => {
    
    const [showsTotal,setShowsTotal] = useState([])
    const [shows,setShows] = useState(showsData)
    const dispatch = useDispatch()
   

    
    useEffect(() => {
        updateShowTotal()
        
    });

    
    const handleToWatchLater = (show) => {
        dispatch(addToWatchLater(show))
    }

    const updateShowTotal = () => {
        setShowsTotal(shows.length)
    }

    const filterGenre = (show) => {
        
        const filterResult = showsData.filter((showData) => {
            return showData.category === show;
        })
        setShows(filterResult)
        
       
    }


   
    
    return (
        <StyledShowsContainer>
            <StyledShowsWrapper>
                <StyledShowsHeader>
                    <StyledShowsTitleWrapper>
                        <StyledShowsTitle>All Shows</StyledShowsTitle>
                        <StyledShowCount>({showsTotal} shows)</StyledShowCount>
                    </StyledShowsTitleWrapper>
                    <StyledShowsRefine>
                        Refine: Popular
                        <StyledShowsRefineIcon>
                            <IoIosArrowDown/>
                        </StyledShowsRefineIcon>
                    </StyledShowsRefine>
                </StyledShowsHeader>
                <StyledShowsGenresWrapper>
                        <StyledShowsGenre onClick={() => filterGenre('Comedy')}>Comedy</StyledShowsGenre>
                        <StyledShowsGenre onClick={() => filterGenre('Documentries')}>Documentries</StyledShowsGenre>
                        <StyledShowsGenre onClick={() => filterGenre('Entertainment')}>Entertainment</StyledShowsGenre>
                        <StyledShowsGenre onClick={() => filterGenre('Film')}>Film</StyledShowsGenre>
                        <StyledShowsGenre onClick={() => filterGenre('Lifestyle')}>Lifestyle</StyledShowsGenre>
                        <StyledShowsGenre onClick={() => filterGenre('News & Current Affairs')}>News & Current Affairs</StyledShowsGenre>
                        <StyledShowsGenre onClick={() => filterGenre('Sports')}>Sports</StyledShowsGenre>
                        <StyledShowsGenre onClick={() => filterGenre('World Drama')}>World Drama</StyledShowsGenre>
                        <StyledShowsGenre onClick={() => filterGenre('On Tv: Yesterday')}>On Tv: Yesterday</StyledShowsGenre>
                        <StyledShowsGenre onClick={() => filterGenre('On Tv: Last 7 Days')}>On Tv: Last 7 Days</StyledShowsGenre>
                        <StyledShowsGenre onClick={() => filterGenre('Box Sets')}>Box Sets</StyledShowsGenre>
                    <StyledShowsMoreBox>
                        More...
                    </StyledShowsMoreBox>
                </StyledShowsGenresWrapper>
                <StyledShows>
                    {shows.map((show)=> {
                        return (
                            <StyledShowsCard onClick={() => handleToWatchLater(show)} key={show.id}>
                                <StyledShowsCardOverlay>
                                    <img src={show.img} alt="show"/>
                                </StyledShowsCardOverlay>
                                <p>{show.name}</p>
                            </StyledShowsCard>
                        )
                        
                    })}
                </StyledShows>
            </StyledShowsWrapper>
        </StyledShowsContainer>
    )
}

export default Shows
