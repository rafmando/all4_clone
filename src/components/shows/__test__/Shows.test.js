import { fireEvent, render } from '@testing-library/react';
import Shows from '../Shows'
import renderer from 'react-test-renderer'
import { store } from '../../../redux/store'
import { Provider } from 'react-redux'

let getByTestId;

const mockShowsData = [
    {id:1, category:'Comedy', name:'Gogglebox', description:"What do Britain's sharpest armchair critics think of our biggest and best TV programmes",img:'/images/gogglebox.jpeg'},
    {id:2, category:'Drama', name:'Hollyoaks', description:"Hollyoaks: the village in Chester where it all happens",img:'/images/hollyoaks.jpeg'},
    {id:3, category:'Entertainment', name:'Made in chelsea', description:"Reality series following the lives, loves and awks of SW3's bright young things",img:'/images/made_in_chelsea.jpeg'},
    {id:4, category:'Entertainment', name:'Married at first sight', description:"Aussie singles get hitched to spouses who they see for the first time on their wedding day",img:'/images/married_at_first_sight.jpeg'},
    {id:5, category:'Documentry', name:'Fraiser', description:"Classic comedy. Dr Frasier Crane is bright and successful. So what could go wrong? Well...",img:'/images/fraiser.jpeg'},
    {id:6, category:'Drama', name:'Close to Me', description:"Drama about a woman who forgets the last year of her life following a brain trauma",img:'/images/close_to_me.jpeg'},
    {id:7, category:'Comedy', name:'Young Sheldon', description:"US comedy prequel to The Big Bang Theory about gifted but eccentric nine-year-old Sheldon Cooper, who doesn't find it easy growing up in East Texas",img:'/images/young_sheldon.jpeg'},
    {id:8, category:'Comedy', name:'The Inbetweeners', description:"Smash-hit adolescent comedy about lads behaving sadly. Welcome to a world of futile crushes, getting drunk too quickly and casting constant aspersions about your friends' sexuality.",img:'/images/the_inbetweeners.jpeg'},
    {id:9, category:'Comedy', name:'Friday Night Dinner', description:"Sitcom from Robert Popper about the gloriously idiosyncratic Goodman family. Starring Simon Bird, Tamsin Greig, Paul Ritter, Tom Rosenthal and guest star Mark Heap.",img:'/images/friday_night_dinner.jpeg'},
    {id:10, category:'Lifestyle', name:'Escape to the Chateau', description:"Can Dick Strawbridge and his wife Angel bring an abandoned French chateau back to life?",img:'/images/escape_to_chateau.jpeg'},
    {id:11, category:'Sports', name:'Formula 1', description:"Welcome to F1 2021. This twisting, turning season is nearing a nail-biting climax, with Max Verstappen and Lewis Hamilton neck and neck. Thrills are guaranteed.",img:'/images/formula_1.jpeg'},
    {id:12, category:'Lifestyle', name:'One Tree Hill', description:"Love, life and basketball in Tree Hill, a small town in North Carolina that sees far more than its fair share of teen drama",img:'/images/one_tree_hill.jpeg'},
]


it("renders the shows data", () => {
    const tree = renderer.create(
        <Provider store={store}>
             <Shows data={mockShowsData}/>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})

it("returns a show when filter button clicked ", () => {
    const filterGenreButton = getByTestId('filter-genre')
    expect(filterGenreButton.textContent).toBe('Comedy')
})
