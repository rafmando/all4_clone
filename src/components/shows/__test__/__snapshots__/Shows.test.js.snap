// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the shows data 1`] = `
<div
  className="sc-bdvvtL gKTgDN"
>
  <div
    className="sc-gsDKAQ hzWZun"
  >
    <div
      className="sc-dkPtRN VKXyE"
    >
      <div
        className="sc-hKwDye ebdpdx"
      >
        <h1
          className="sc-jRQBWg isoPMB"
        >
          All Shows
        </h1>
        <p
          className="sc-eCImPb hUVamz"
        >
          (
           shows)
        </p>
      </div>
      <div
        className="sc-gKclnd gspfhA"
      >
        Refine: Popular
        <div
          className="sc-iCfMLu kLEZTn"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 294.1L383 167c9.4-9.4 24.6-9.4 33.9 0s9.3 24.6 0 34L273 345c-9.1 9.1-23.7 9.3-33.1.7L95 201.1c-4.7-4.7-7-10.9-7-17s2.3-12.3 7-17c9.4-9.4 24.6-9.4 33.9 0l127.1 127z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="sc-furwcr lcJmFS"
    >
      <div
        className="sc-pVTFL dFQlVp"
        onClick={[Function]}
      >
        Comedy
      </div>
      <div
        className="sc-pVTFL dFQlVp"
        onClick={[Function]}
      >
        Documentries
      </div>
      <div
        className="sc-pVTFL dFQlVp"
        onClick={[Function]}
      >
        Entertainment
      </div>
      <div
        className="sc-pVTFL dFQlVp"
        onClick={[Function]}
      >
        Film
      </div>
      <div
        className="sc-pVTFL dFQlVp"
        onClick={[Function]}
      >
        Lifestyle
      </div>
      <div
        className="sc-pVTFL dFQlVp"
        onClick={[Function]}
      >
        News & Current Affairs
      </div>
      <div
        className="sc-pVTFL dFQlVp"
        onClick={[Function]}
      >
        Sports
      </div>
      <div
        className="sc-pVTFL dFQlVp"
        onClick={[Function]}
      >
        World Drama
      </div>
      <div
        className="sc-pVTFL dFQlVp"
        onClick={[Function]}
      >
        On Tv: Yesterday
      </div>
      <div
        className="sc-pVTFL dFQlVp"
        onClick={[Function]}
      >
        On Tv: Last 7 Days
      </div>
      <div
        className="sc-pVTFL dFQlVp"
        onClick={[Function]}
      >
        Box Sets
      </div>
      <div
        className="sc-jrQzAO hqnedw"
      >
        More...
      </div>
    </div>
    <div
      className="sc-kDTinF iOqFMu"
    >
      <div
        className="sc-iqseJM itDdmz"
        onClick={[Function]}
      >
        <div
          className="sc-crHmcD XkmdP"
        >
          <img
            alt="show"
            src="/images/gogglebox.jpeg"
          />
        </div>
        <p>
          Gogglebox
        </p>
      </div>
      <div
        className="sc-iqseJM itDdmz"
        onClick={[Function]}
      >
        <div
          className="sc-crHmcD XkmdP"
        >
          <img
            alt="show"
            src="/images/hollyoaks.jpeg"
          />
        </div>
        <p>
          Hollyoaks
        </p>
      </div>
      <div
        className="sc-iqseJM itDdmz"
        onClick={[Function]}
      >
        <div
          className="sc-crHmcD XkmdP"
        >
          <img
            alt="show"
            src="/images/made_in_chelsea.jpeg"
          />
        </div>
        <p>
          Made in chelsea
        </p>
      </div>
      <div
        className="sc-iqseJM itDdmz"
        onClick={[Function]}
      >
        <div
          className="sc-crHmcD XkmdP"
        >
          <img
            alt="show"
            src="/images/married_at_first_sight.jpeg"
          />
        </div>
        <p>
          Married at first sight
        </p>
      </div>
      <div
        className="sc-iqseJM itDdmz"
        onClick={[Function]}
      >
        <div
          className="sc-crHmcD XkmdP"
        >
          <img
            alt="show"
            src="/images/fraiser.jpeg"
          />
        </div>
        <p>
          Fraiser
        </p>
      </div>
      <div
        className="sc-iqseJM itDdmz"
        onClick={[Function]}
      >
        <div
          className="sc-crHmcD XkmdP"
        >
          <img
            alt="show"
            src="/images/close_to_me.jpeg"
          />
        </div>
        <p>
          Close to Me
        </p>
      </div>
      <div
        className="sc-iqseJM itDdmz"
        onClick={[Function]}
      >
        <div
          className="sc-crHmcD XkmdP"
        >
          <img
            alt="show"
            src="/images/young_sheldon.jpeg"
          />
        </div>
        <p>
          Young Sheldon
        </p>
      </div>
      <div
        className="sc-iqseJM itDdmz"
        onClick={[Function]}
      >
        <div
          className="sc-crHmcD XkmdP"
        >
          <img
            alt="show"
            src="/images/the_inbetweeners.jpeg"
          />
        </div>
        <p>
          The Inbetweeners
        </p>
      </div>
      <div
        className="sc-iqseJM itDdmz"
        onClick={[Function]}
      >
        <div
          className="sc-crHmcD XkmdP"
        >
          <img
            alt="show"
            src="/images/friday_night_dinner.jpeg"
          />
        </div>
        <p>
          Friday Night Dinner
        </p>
      </div>
      <div
        className="sc-iqseJM itDdmz"
        onClick={[Function]}
      >
        <div
          className="sc-crHmcD XkmdP"
        >
          <img
            alt="show"
            src="/images/escape_to_chateau.jpeg"
          />
        </div>
        <p>
          Escape to the Chateau
        </p>
      </div>
      <div
        className="sc-iqseJM itDdmz"
        onClick={[Function]}
      >
        <div
          className="sc-crHmcD XkmdP"
        >
          <img
            alt="show"
            src="/images/formula_1.jpeg"
          />
        </div>
        <p>
          Formula 1
        </p>
      </div>
      <div
        className="sc-iqseJM itDdmz"
        onClick={[Function]}
      >
        <div
          className="sc-crHmcD XkmdP"
        >
          <img
            alt="show"
            src="/images/one_tree_hill.jpeg"
          />
        </div>
        <p>
          One Tree Hill
        </p>
      </div>
    </div>
  </div>
</div>
`;
